trolley/offline_payment_gateway.py

```diff
@@ -0,0 +1,93 @@
+            endpoint = f'/v1/offline-payments?page={page}&pageSize={pageSize}'
+            response = trolley.configuration.Configuration.client(self.config).get(endpoint)
+            return self.__build_offline_payments_from_response(response, True)
+
+    """ Creates a new Offline Payment for a Recipient whose ID is provided """
+    def create(self, recipient_id, body,):
+        if body is None:
+            raise InvalidFieldException("Body cannot be None.")
+        elif recipient_id is None:
+            raise InvalidFieldException("Recipient ID cannot be None.")
+        endpoint = f'/v1/recipients/{recipient_id}/offlinePayments/'
+        response = trolley.configuration.Configuration.client(
+            self.config).post(endpoint, body)
+
+        temppayment = trolley.OfflinePayment.factory(response)
+        offlinepayment = namedtuple("OfflinePayment", temppayment.keys())(*temppayment.values())
+        return offlinepayment
+
+    """ Update an Offline Payment """
+    def update(self, offline_payment_id, recipient_id, body):
+        if offline_payment_id is None:
+            raise InvalidFieldException("Offline Payment id cannot be None.")
+        if body is None:
+            raise InvalidFieldException("Body cannot be None.")
+        if recipient_id is None:
+            raise InvalidFieldException("Recipient ID cannot be None.")
+        endpoint = f'/v1/recipients/{recipient_id}/offlinePayments/{offline_payment_id}'
+        response = trolley.configuration.Configuration.client(
+            self.config).patch(endpoint, body)
+        temppayment = trolley.OfflinePayment.factory(response)
+        offlinepayment = namedtuple("OfflinePayment", temppayment.keys())(*temppayment.values())
+        return offlinepayment
+
+    """ Delete an Offline Payment """
+    def delete(self, recipient_id, offline_payment_id):
+        if recipient_id is None:
+            raise InvalidFieldException("Recipient ID cannot be None.")
+        if offline_payment_id is None:
+            raise InvalidFieldException("Offline Payment ID cannot be None.")
+        
+        endpoint = f'/v1/recipients/{recipient_id}/offlinePayments/{offline_payment_id}'
+        response = trolley.configuration.Configuration.client(
+            self.config).delete(endpoint)
+        return response['ok']
+        
+    def __build_offline_payments_from_response(self, response, include_meta=False):
+        offlinepayments = []
+        count = 0
+        for offlinepayment in response['offlinePayments']:
+            temppayment = trolley.OfflinePayment.factory(offlinepayment)
+            newpayment = namedtuple("OfflinePayment", temppayment.keys())(*temppayment.values())
+            offlinepayments.insert(count, newpayment)
+            count = count + 1
+        
+        if include_meta:
+            tempmeta = Meta.factory(response['meta'])
+            offlinepayments.insert(count,namedtuple("Meta", tempmeta.keys())(*tempmeta.values()))
```

Comment:
Same concern on insert. in the next loop, when you do offlinePayments.insert(count, newpayment), I am afraid it will overwrite the meta. is it on purpose?

Reply:
oh yes, it is out of the for loop:)

Reply:
@julie-trolley so can we approve this one now? :D